orderbook/src/database/orders.rs

```diff
@@ -305,8 +311,14 @@ mod tests {
             .await
             .unwrap();
         assert_eq!(db_orders[0].order_meta_data.invalidated, true);
-        // TODO - cancel twice and verify that first cancellation date isn't over written
-        // This will require querying the DB for the cancellation_date.
+
+        let query = "SELECT cancellation_timestamp FROM orders;";
```

Comment:
nit: a slightly better way to test this is instead of using `Utc::now()` in the database implementation, we inject our own abstract `now` function that is is using `Utc::now()` in production, but that we can mock in our unit tests.

An example can be found here: https://github.com/gnosis/oba-services/blob/b6de9c3b3193d6e23b217a2442428429a54e690c/orderbook/src/fee.rs#L20

and a test that manipulates the time: https://github.com/gnosis/oba-services/blob/b6de9c3b3193d6e23b217a2442428429a54e690c/orderbook/src/fee.rs#L292

Otherwise this test might not be catching issues on a very fast machine where both inserts happen in the same millisecond.

Reply:
Very good point and thanks for the pointers here. My assumption then would be to include this now field on the `OrderCancellation` struct so that it can be used in the query? Or is the a more appropriate place for this?

Reply:
I am beginning to think this change might be a more involved refactoring of how we use time on the entire database. If its alright with you I might suggest leaving this change for a separate PR.

Reply:
Maybe a lot simpler than using a trait would be to just pass the current time `now` into `cancel_order` as a parameter instead of creating it there. Should only be a  2 line change.

Reply:
Actually, I think I may have found something that works by implementing an `async now` method on Database that can be used as `Database::now().await`. Not sure if this is what you had in mind, but please let me know your thoughts.

Reply:
Thanks @vkgnosis - I will go with that.

Reply:
I meant changing

```
 pub async fn cancel_order(&self, order_uid: &OrderUid) -> Result<()>
```

to 

```
 pub async fn cancel_order(&self, order_uid: &OrderUid, now: DateTime<Utc>) -> Result<()>
```

That way who ever calls the function has control over the time. In the test you can then manually increment the time while in the orderbook you would set the parameter to `Utc::now()`.

Reply:
Awesome. This looks much better.